### ISSUE TITLE
SMCSuperIO and fan and other sensor data for Z590

### ISSUE THREAD

**Issue by stiligFox** [acidanthera/bugtracker#1666]
I have a Gigabyte Z590 Vision G with an i9-10900k; In my in my list of sensors in iStat menus, I don't have any fans or all of the available temperature sensors for this board available, as I only have the sensors for the first 4 cores, and I'm missing about three other motherboard temp sensors. The fans in my build also don't seem to follow the BIOS defined fan curves; no matter how I bench test the system, the fans don't ramp up or down at all. I am running all the most recent kexts and OpenCore 0.6.9.

![Screen Shot 2021-06-05 at 1 51 47 PM](https://user-images.githubusercontent.com/28546763/120902300-2f8f7900-c605-11eb-92f6-e5d3e01e72c7.png)

I know nothing about programing, but if there is anything I can do to help, any information I can grab that might help, I'd love to!

On a related note, I also don't have speeds or temps for my Sapphire 5700XT Nitro +; I don't know if there's anything that can be done about that as well?

**Comment by vit9696** [comment]
Check HWMonitorSMC2, if it shows nothing you are likely out of luck. If it shows something, bug iStat menus author.

**Comment by 1alessandro1** [comment]
@stiligFox what SuperIO chip do you have onboard? Usually gigabyte motherboards have ITE chips, what's on yours? 

You can use Linux and `superiotool` if you're unable to see the chip name manually

**Comment by stiligFox** [comment]
@1alessandro1 I ran superiotool in Ubuntu, but superiotool r6637 states “No Super I/O found.” It is listed in my motherboard manual that there is an ITE device, though not which one.

**Comment by 1alessandro1** [comment]
@stiligFox your option is to check if you can find full datasheets for those two devices, or email ITE asking for them (the latter might be the only one since this motherboard is very new)

**Comment by stiligFox** [comment]
@1alessandro1 so I know it’s been a long time, but I finally got to write ITE and they told me both those chips are “customer design IC.
So we cannot provide related information to you.” I suppose that makes sense, but that stops any chance of adding sensor data dead in the water…

Are there any other options? I know the sensor data for these were added to HWInfo a while back for windows, so sure the data sheets are out there somewhere? Would I contact Gigabyte?

**Comment by zhen-zen** [comment]
Recall a piece of news earlier this year on Gigabyte WMI driver: https://www.phoronix.com/scan.php?page=news_item&px=Gigabyte-WMI-Linux-Temperature

It's merged into Linux mainstream now but this repo still provides some examples for reference: https://github.com/t-8ch/linux-gigabyte-wmi-driver

I just implemented similar WMI based sensor reading for my mini PC when I read that. So I checked their repo and parsed the descriptor in the SSDT just for curiosity. But I got quite busy in life this summer and not able to support a hardware I don't have like the ThinkPad case last year. It should fit in the same framework of YogaSMC though.
